jdaviz/configs/mosviz/helper.py

```diff
@@ -528,19 +579,180 @@ def load_images(self, data_obj, data_labels=None, share_image=0):
+            raise TypeError("column_name must be of type str")
+
+        column_names = self.get_column_names(True)
+        if column_name not in column_names:
+            if column_name in self.get_column_names():
+                return self.set_visible_columns(column_names+[column_name])
+            else:
+                raise ValueError(f"{column_name} not in available columns ({column_names})")
+
+    def get_column(self, column_name):
+        """
+        Get the data from a column in the table.
+
+        Parameters
+        ----------
+        column_name: str
+            Header string of an existing column in the table.
+
+        Returns
+        -------
+        array
+            copy of the data array.
         """
-        Add a new data column to the table.
+        return np.asarray(deepcopy(self.app.data_collection['MOS Table'].get_component(column_name).data)) # noqa
+
+    def add_column(self, column_name, data=None, show=True):
+        """
+        Add a new data column to the table or update the data in an existing column.
+
+        If ``column_name`` is 'Redshift', the column will be synced with the redshift
+        in the respective spectrum objects.
 
         Parameters
         ----------
+        column_name : str
+            Header string to be shown in the table.  If already exists as a column in
+            the table, the data for that column will be updated.
         data : array-like
             Array-like set of data values, e.g. redshifts, RA, DEC, etc.
-        column_name : str
-            Header string to be shown in the table.
+        show: bool or None
+            Whether to show the column in the table (defaults to True).  If None, will
+            show if the column is new, otherwise will leave at current state.
+
+        Returns
+        -------
+        array
+            copy of the data in the added or edited column.
         """
         table_data = self.app.data_collection['MOS Table']
-        table_data.add_component(data, column_name)
+
+        if data is None:
+            data = [None]*table_data.size
+        if not isinstance(data, (list, tuple, np.ndarray)):
+            raise TypeError("data must be array-like")
+        if len(data) != table_data.size:
+            raise ValueError(f"data must have length {table_data.size} (rows in table)")
+
+        if column_name in self.get_column_names():
+            table_data.update_components({table_data.get_component(column_name): data})
+        else:
+            table_data.add_component(data, column_name)
+        if show is True:
+            self.show_column(column_name)
+        elif not show and show is not None:
+            self.hide_column(column_name)
+
+        if column_name == 'Redshift':
+            # apply the value in the current row to the specviz object
+            self._apply_redshift_from_table()
+
+        return self.get_column(column_name)
+
+    def update_column(self, column_name, data, row=None):
```

Comment:
Also, this method allows a call like this. I thought it would crash, but it didn't.

```python
mosviz.update_column('Redshift', data=[0, 1, 2, 3, 'asasddas'])
```

Reply:
p.s. One of the comment disappeared after I posted the review...

Maybe also a method to rename column without changing the data? Or is that too chaotic?

Reply:
These are good ideas, but I am a bit worried we're just bloating and adding a lot of calls that won't be used/tested unless we add UI capabilities to do all of this.  Should we implement this now or defer (its easy either way)?

Reply:
Defer is okay.

Reply:
p.s. I am okay with deferring `delete_column` and `rename_column` but I think it is still important to check for valid redshift values as part of this PR, no? I posted a comment outside of this thread on the possible way it can break if we do not check for redshift validity beforehand.